app/services/data_migrations/backfill_temp_sites.rb

```diff
@@ -0,0 +1,51 @@
+module DataMigrations
+  class BackfillTempSites
+    TIMESTAMP = 20220517111432
+    MANUAL_RUN = false
+
+    def change
+      CycleTimetable::CYCLE_DATES.each_key do |year|
+        next if year == CycleTimetable.next_year
+
+        Provider.all.each do |provider|
+          migrate_temp_sites_for_courses(provider, year)
+        end
+      end
+    end
+
+  private
+
+    def migrate_temp_sites_for_courses(provider, year)
+      courses = TeacherTrainingPublicAPI::Course.where(
+        year: year,
+        provider_code: provider.code,
+      ).paginate(per_page: 500)
+
+      courses.each do |course|
+        migrate_temp_sites_for_course(provider, course, year)
+      end
+    end
+
+    def migrate_temp_sites_for_course(provider, course, year)
+      sites = TeacherTrainingPublicAPI::Location.where(
+        year: year,
+        provider_code: provider.code,
+        course_code: course.code,
+      ).includes(:location_status).paginate(per_page: 500)
+
+      sites.each do |site|
+        temp_site = TeacherTrainingPublicAPI::AssignTempSiteAttributes.new(site, provider).call
+        temp_site.save!
+        attach_to_course_options(temp_site)
+      end
+    end
+
+    def attach_to_course_options(temp_site)
+      return unless (matching_site = Site.find_by(provider: temp_site.provider, code: temp_site.code))
+
+      matching_site.course_options.each do |course_option|
```

Comment:
A site can have multiple course options for courses across multiple years. Since we are scoping `temp_site` to a single year we cannot loop through course options for a site as they will include all recruitment cycle years. Could we instead find the course option using the specific course id for that year, site id and study mode (similar to the sync site) and attach the temp site to it instead? This way we avoid attaching course options to temp sites across different years

Reply:
Will do, thanks